homeassistant/components/ping/binary_sensor.py

```diff
@@ -123,13 +124,15 @@ def __init__(self, host, count):
                 self._ip_address,
             ]
 
-    def ping(self):
+    async def async_ping(self):
         """Send ICMP echo request and return details if success."""
-        pinger = subprocess.Popen(
-            self._ping_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        pinger = await asyncio.create_subprocess_shell(
+            " ".join(self._ping_cmd),
```

Comment:
Should we store the string in `self._ping_cmd` from the beginning so we don't need to convert it every ping?

Reply:
right. this should be fixed now.